client/src/components/AddCat/AddCat.tsx

```diff
@@ -0,0 +1,77 @@
+import { useState, useCallback } from "react";
+import { useStyles } from './style';
+import { useCats } from "../../contexts/cats.context";
+import { Button } from "../common/components/Button";
+import { useNavigate } from "react-router-dom";
+
+const ADD_CAT_TEXT = 'Add Cat';
+const DESCRIPTION_TEXT = 'Description';
+const IMAGE_TEXT = 'Image';
+const LAST_NAME_TEXT = 'Last Name';
+const FIRST_NAME_TEXT = 'First Name';
+
+export const AddCat = () => {
+    const [firstName, setFirstName] = useState<string>('');
+    const [lastName, setLastName] = useState<string>('');
+    const [image, setImage] = useState<string>('');
+    const [description, setDescription] = useState<string>('');
+    const navigate = useNavigate();
+
+    const styles = useStyles();
+    const store = useCats();
+
+    const handleFirstNameChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+        setFirstName(event.target.value);
+    }, []);
+
+    const handleLastNameChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+        setLastName(event.target.value);
+    }, []);
+
+    const handleImageChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+        setImage(event.target.value);
+    }, []);
+
+    const handleDescriptionChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+        setDescription(event.target.value);
+    }, []);
+
+    const handleSubmit = useCallback((event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        const newCat = {
+            firstName: firstName,
+            lastName: lastName,
+            image: image,
+            description: description,
+            mice: []
+        }
+        store.actions.addCat(newCat);
+        navigate('/');
```

Comment:
why not await the `addCat` function before navigation? right now if the navigation is a bit faster than the creation I will not see my new one